src/browser/modules/D3Visualization/lib/visualization/components/Visualization.ts

```diff
@@ -230,12 +235,42 @@ export class Visualization {
     this.forceSim.updateRelationships(this.graph)
   }
 
-  private zoomClick(isZoomingIn: boolean): void {
+  private handleZoomClick(isZoomingIn: boolean): void {
     this.draw = true
     this.isZoomClick = true
     this.zoomBehavior.scaleBy(this.root, isZoomingIn ? 1.3 : 0.7)
   }
 
+  private handleZoomToFitClick(): void {
+    this.draw = true
+    this.isZoomClick = true
+    this.zoomToFitWholeGraph()
+  }
+
+  private zoomToFitWholeGraph = () => {
```

Comment:
Considering that they're always used together, wouldn't it make sense to combine `handleZoomToFitClick`, `zoomToFitWholeGraph` and `zoomToFitClick` into one function?

Reply:
I see what you mean, however I like that it is clear connections to the corresponding zoom in/out functions and the common stuff needed to be done, so I will reduce the functions and make that connection clearer by combining those three clicks into one handler.